components/tag-list/tag-list-item-mixin.js

```diff
@@ -71,24 +108,71 @@ export const TagListItemMixin = superclass => class extends superclass {
 	firstUpdated(changedProperties) {
 		super.firstUpdated(changedProperties);
 
-		const container = this.shadowRoot.querySelector('.tag-list-item-container');
+		const container = this.shadowRoot.querySelector('.tag-list-item-content');
 		this.addEventListener('focus', (e) => {
 			// ignore focus events coming from inside the tag content
-			if (e.composedPath()[0] === this) container.focus();
+			if (e.composedPath()[0] !== this) return;
+			const event = new FocusEvent('focus', { bubbles: true, cancelable: true });
+			container.dispatchEvent(event);
+		});
+		this.addEventListener('blur', () => {
+			const event = new FocusEvent('blur', { bubbles: true, cancelable: true });
+			container.dispatchEvent(event);
 		});
-		this.addEventListener('blur', () => container.blur());
+		this.addEventListener('keydown', this._handleKeydown);
+	}
+
+	deleteItem() {
+		this.parentNode.removeChild(this);
+	}
+
+	handleClearItem(e) {
+		if (!this.clearable) return;
+
+		let handleFocus = false;
+		if (e) {
+			const listItemParent = findComposedAncestor(e.composedPath()[0], (node) => node.role === 'listitem');
+			handleFocus = listItemParent ? true : false;
+		}
+		/** Dispatched when a user selects to delete a tag list item. The consumer must handle the actual element deletion. */
+		this.dispatchEvent(new CustomEvent(
+			'd2l-tag-list-item-cleared',
+			{ bubbles: true, composed: true, detail: { value: this.text, handleFocus } }
+		));
+	}
+
+	_handleKeydown(e) {
+		const expectedKey = e.keyCode === 8 || e.keyCode === 46; // backspace or delete
+		if (!this.clearable || !expectedKey) return;
+		e.preventDefault();
+		this.handleClearItem(e);
 	}
 
 	_renderTag(tagContent, hasTruncationTooltip) {
+		const buttonText = typeof tagContent === 'object'
+			? this.localize('components.tag-list.clear', { value: '' })
+			: this.localize('components.tag-list.clear', { value: tagContent });
 		const tooltip = hasTruncationTooltip ? html`
-				<d2l-tooltip for="${this._id}" show-truncated-only>
+				<d2l-tooltip for="${this._id}" offset="20" show-truncated-only>
```

Comment:
This offset is because the div is now within the container so it needs some cushion. This isn't ideal if the tooltip were to open to the right when clearable. It seems unlikely but is possible depending on layout.